diff --git a/pbkzlib/Makefile b/pbkzlib/Makefile
index 4a9db83..22c16cc 100644
--- a/pbkzlib/Makefile
+++ b/pbkzlib/Makefile
@@ -1,11 +1,11 @@
-CC	= g++
-SRC	= main.cpp
-BOOST_DIR = ./boost_1_66_0
-CFLAGS	= -std=c++11 -march=native -Ofast -funroll-loops -flto -fomit-frame-pointer -fprefetch-loop-arrays -msse4 -fpermissive
-LDFLAGS	= -lgmp -lmpfr -lgsl -lntl -fopenmp
-INCL	= -I. -I${BOOST_DIR}
-
-all:
-	${CC} ${CFLAGS} ${SRC} ${INCL} ${LDFLAGS}
+include ../common.mk
 
+CC	= $(CXX)
+INCL	= -I. -I$(ROOT_DIR)/prefix/include
+CFLAGS	= -std=c++14 -march=native -Ofast -funroll-loops -flto -fomit-frame-pointer -fprefetch-loop-arrays -msse4 -fpermissive -Wall -Wextra $(INCL)
+CXXFLAGS = $(CFLAGS)
+LDFLAGS	=  -L$(ROOT_DIR)/prefix/lib -fopenmp
+LOADLIBES = -lgmp -lmpfr -lgsl -lgslcblas -lntl
 
+main.out: main.cpp
+	$(LINK.cpp) $^ $(LOADLIBES) $(LDLIBS) -o $@
